src/refmt_impl.ml

```diff
@@ -35,19 +37,34 @@ let reasonBinaryParser use_stdin filename =
   let (magic_number, filename, ast, comments, parsedAsML, parsedAsInterface) = input_value chan in
   ((ast, comments), parsedAsML, parsedAsInterface)
 
-let ocamlBinaryParser use_stdin filename parsedAsInterface =
-  let chan =
-    match use_stdin with
-      | true -> stdin
-      | false ->
-          let file_chan = open_in filename in
-          seek_in file_chan 0;
-          file_chan
+let ocamlBinaryParser use_stdin filename =
+  let chan = match use_stdin with
+    | true -> stdin
+    | false ->
+      let file_chan = open_in filename in
+      seek_in file_chan 0;
+      file_chan
   in
-  let _ = really_input_string chan (String.length Config.ast_impl_magic_number) in
-  let _ = input_value chan in
-  let ast = input_value chan in
-  ((ast, []), true, parsedAsInterface)
+  match Ast_io.from_channel chan with
```

Comment:
Does this even include logic for marshaling between two *OCaml* runtime versions? I don't think it does but that would be great because then we could check in parsed ASTs as artifacts in the package manager - which allows us to remove `reason` as a dependency of packages that use `reason` syntax. I think that would require `migrate-parse-tree` to depend on `n` versions of the OCaml runtime implementation (basically three OCaml languages) as a dependency of `migrate-parse-tree`.